pkg/webhook/server.go

```diff
@@ -369,34 +373,81 @@ func (wh *webhook) handleWebhookValidate(w http.ResponseWriter, req *http.Reques
 	)
 }
 
-func reviewResponse(uid types.UID, err error, aletmanagerHost string, resource string, name string, namespace string) *admissionv1.AdmissionReview {
-	// Currently hard coded and commented logic in order to just audit, but in order to get the content we need the policy to be deny.
-	// allowed := err == nil
-	//var status int32 = http.StatusAccepted
-	// if err != nil {
-	// 	status = http.StatusForbidden
-	// }
+func getValidationAnnotations(attrs admission.Attributes) (audit bool, deny bool) {
+	validationActionsPattern := `validationActions":\[(.*?)\]`
+	regex, _ := regexp.Compile(validationActionsPattern)
+
+	match := regex.FindStringSubmatch(fmt.Sprintf("%+v", attrs))
+	if len(match) >= 2 {
+		actions := match[1]
+		audit = strings.Contains(actions, "Audit")
+		deny = strings.Contains(actions, "Deny")
+	}
+
+	logger.Info("The actions are", "audit", audit, "deny", deny)
+
+	return audit, deny
+}
+
+func getMessage(attrs admission.Attributes) (message string) {
+	validationMessagePattern := `message":"(.*?)"`
+	regex, _ := regexp.Compile(validationMessagePattern)
+
+	match := regex.FindStringSubmatch(fmt.Sprintf("%+v", attrs))
+	if len(match) >= 2 {
+		message = match[1]
+	}
+
+	logger.Info("The message is", "message", message)
+
+	return message
+}
+
+func getPolicy(attrs admission.Attributes) (policy string) {
+	policyPattern := `policy":"(.*?)"`
+	regex, _ := regexp.Compile(policyPattern)
+
+	match := regex.FindStringSubmatch(fmt.Sprintf("%+v", attrs))
+	if len(match) >= 2 {
+		policy = match[1]
+	}
+
+	logger.Info("The policy is", "policy", policy)
+
+	return policy
+}
```

Comment:
What if you don't find? I mean there is no audit action (policy failure) I guess we would not find this pattern.

Reply:
Then the policy is "" which is fine because I use it only in case audit || policy is true